samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/ui/settings/Settings.kt

```diff
@@ -1,9 +1,305 @@
+            color = MaterialTheme.colorScheme.primary
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun EnvSettingsList(
+    deviceToken: String,
+    trackUrl: String,
+    onTrackUrlChange: (trackUrl: String) -> Unit
+) {
+    var trackUrlState by remember { mutableStateOf(trackUrl) }
+
+    Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth(),
+            value = deviceToken,
+            readOnly = true,
+            onValueChange = {},
+            label = {
+                Text(text = stringResource(id = R.string.device_token))
+            }
+        )
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth(),
+            value = trackUrlState,
+            onValueChange = {
+                trackUrlState = it
+                onTrackUrlChange(it)
+            },
+            label = {
+                Text(text = stringResource(id = R.string.cio_track_url))
+            },
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Uri)
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun WorkspaceSettingsList(
+    uiState: SettingsUiState,
+    onSiteIdChange: (siteId: String) -> Unit,
+    onApiKeyChange: (apiKey: String) -> Unit
+) {
+    var siteId by remember { mutableStateOf(uiState.configuration.siteId) }
+    var apiKey by remember { mutableStateOf(uiState.configuration.apiKey) }
+
+    Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
+        OutlinedTextField(modifier = Modifier.fillMaxWidth(), value = siteId, onValueChange = {
+            siteId = it
+            onSiteIdChange(it)
+        }, label = {
+            Text(text = stringResource(id = R.string.site_id))
+        })
+        OutlinedTextField(modifier = Modifier.fillMaxWidth(), value = apiKey, onValueChange = {
+            apiKey = it
+            onApiKeyChange(it)
+        }, label = {
+            Text(text = stringResource(id = R.string.api_key))
+        })
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SDKSettingsList(
+    uiState: SettingsUiState,
+    onBackgroundQueueSecondsDelayChange: (delay: Double) -> Unit,
+    onBackgroundQueueMinNumTasksChange: (numTasks: Int) -> Unit
+) {
+    var secondsDelays by remember { mutableStateOf(uiState.configuration.backgroundQueueSecondsDelay) }
+    var minTasks by remember { mutableStateOf(uiState.configuration.backgroundQueueMinNumTasks) }
+
+    Column(verticalArrangement = Arrangement.spacedBy(4.dp), modifier = Modifier.fillMaxWidth()) {
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth(),
+            value = secondsDelays.toString(),
+            onValueChange = {
+                if (it.isNotEmpty()) {
+                    secondsDelays = it.toDouble()
+                    onBackgroundQueueSecondsDelayChange(it.toDouble())
+                }
```

Comment:
Shouldn't we use default value if it is empty? e.g.

1. Change the value to `10`
2. Save settings
3. Go to settings screen again
4. Remove the value completely
5. Save settings

Will the value saved be `10` (last saved) or `30` (default value) in this case? 🤓

Reply:
It makes it 0 if you remove it, so when you save it its saved as 0.